### ISSUE TITLE
`private_unit_test` false positive for private `XCTestCase` subclass

### ISSUE THREAD

**Issue by mattrubin** [realm/SwiftLint#3277]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The `private_unit_test` rule incorrectly warns "Unit test marked `private` will not be run by XCTest" for an `XCTestCase` subclass marked as `private`. With Xcode 11.5, XCTest _will_ run test methods for a private `XCTestCase` subclass.

```swift
import XCTest

private class ExampleTests: XCTestCase {
    func testExample() throws {
        print("This test runs!")
    }
}
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Linting Swift files at paths 
Linting 'private_unit_testTests.swift' (1/4)
Linting 'SceneDelegate.swift' (2/4)
Linting 'ContentView.swift' (3/4)
Linting 'AppDelegate.swift' (4/4)
/Users/mattrubin/private_unit_test/private_unit_testTests/private_unit_testTests.swift:11:9: warning: Private Unit Test Violation: Unit test marked `private` will not be run by XCTest. (private_unit_test)
Done linting! Found 1 violation, 0 serious in 4 files.
```

### Environment

* SwiftLint version: **0.39.2**
* Installation method used: **Homebrew**
* Paste your configuration file: **N/A (using the default configuration)**
* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations): **No**
* Which Xcode version are you using (check `xcodebuild -version`)? **Xcode 11.5, Build version 11E608c**
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
private class ExampleTests: XCTestCase {
    func testExample() throws {
        print("This test runs!")
    }
}
```

**Comment by jpsim** [comment]
XCTest must have added support for this since the rule was originally written. If you can identify what version of Swift corresponded with the support, we could make the rule no-op if the detected Swift version matches that or is higher.
